src/or/circuitlist.c

```diff
@@ -625,7 +625,7 @@ circuit_any_opened_circuits(void)
         TO_CIRCUIT(next_circ)->state == CIRCUIT_STATE_OPEN &&
         TO_CIRCUIT(next_circ)->purpose != CIRCUIT_PURPOSE_C_MEASURE_TIMEOUT &&
         next_circ->build_state &&
-        next_circ->build_state->desired_path_len == DEFAULT_ROUTE_LEN) {
+        next_circ->build_state->desired_path_len >= DEFAULT_ROUTE_LEN) {
```

Comment:
How does that work with single onion service circuits? I'm not entirely sure, because of preemptive circuit building, if a tor client can NOT have at least 3-hop circuits but I'm guessing their could be a window that it can happen?

Reply:
Hrmm yeah. Ok. We should count SoS circuits here, too. Tried to do that in this commit: 8c950e73d302a6f0b5ada126a5aeac55990f0cc8